web3-dev/smart-contracts/getting-started.rst

```diff
@@ -84,117 +86,141 @@ Note that a `deployer.wasm` file has also been generated. It will be used to dep
-   :ref:`Local network generation <local-network-generation>`.
+   import { Address, call } from '@massalabs/massa-as-sdk';
+   import { Args } from '@massalabs/as-types';
 
-Execute the smart contract on the node
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+   export function constructor(args: StaticArray<u8>): StaticArray<u8> {
+      callHelloContract(args);
+      return [];
+   }
 
-Everything is in place, we can now execute the `hello world` smart contract on your
-local node with the following command inside the **client cli**:
+   function callHelloContract(args: StaticArray<u8>): void {
+      const address = new Address(
+         '<your contract address>',
+      );
+      call(address, 'sayHello', new Args(args), 0);
+      return;
+   }
 
-.. code-block:: shell
+* line 4: `constructor` function is a special function which will be called when our run smart-contract will be deployed.
+* line 9: `callHelloContract` function will initialize an Address object containing the deployed smart-contract address and call it.
+* line 13: `call` function makes the call to an already deployed contract, at the given address.
 
-   send_smart_contract <address> <path to wasm file> 100000 0 0
+Now that everything is ready, we have to build our new contract:
 
 .. note::
+   * The `npm run build` command will build every smart contract in the folder `assembly/contracts`.
 
-   We are executing the send_smart_contract command with 6 parameters:
+.. code-block:: shell
 
-   - <address>: the address of your wallet kept during previous step;
-   - <path to wasm file>: the full path (from the root directory to the file extension .wasm)
-     of the hello smart contract, generated in the previous chapter.
-   - 100000: the maximum amount of gas that the execution of your smart contract is allowed to use.
-   - Two 0 parameters that can be safely ignored by now. If you want more info on them, use the command
-     `help send_smart_contract`.
+   npm run build
 
-.. note::
+Now let's have a look at the deployer script `src/deployer.ts` and stop at the `deploySC` instruction.
 
-   To go inside the **client cli**, open a terminal in `massa/massa-client` directory and run `cargo run`.
+.. code-block:: typescript
 
-If everything went fine, the following message should be prompted:
+  ...
+  await deploySC(
+    publicApi,
+    deployerAccount,
+    [
+      {
+        data: readFileSync(path.join(__dirname, 'build', 'main.wasm')),
+        coins: 0,
+        args: new Args().addString('test'),
+      } as ISCData,
+    ],
+    0,
+    4_200_000_000,
+    true,
+  );
+  ...
+
+This function lets us set the smart contract to be deployed when running `npm run deploy`.
+It also allows us to add arguments to the call.
+Let's modify the code in order to deploy our run smart contract by replacing `main.wasm` by `run.wasm`.
+We also want to set a name as argument to be used in our `sayHello` contract by replacing `test` by your name!
```

Comment:
```suggestion
Let's also replace the `test` string used as an argument by our name so that it will be passed to the `sayHello` function of our smart contract!
```